java/client/src/org/openqa/selenium/chrome/ChromeOptions.java

```diff
@@ -89,9 +76,78 @@ public ChromeOptions() {
   @Override
   public ChromeOptions merge(Capabilities extraCapabilities) {
     super.merge(extraCapabilities);
+    if (extraCapabilities instanceof ChromeOptions) {
+      Map options = (Map) extraCapabilities.asMap().get("goog:chromeOptions");
+      mergeArguments(options);
+      mergeExtensions(options);
+      mergeExtensionFiles(extraCapabilities);
+      mergeBinary(options);
+      mergeExperimentalOptions(extraCapabilities);
+    }
     return this;
   }
 
+  private void mergeArguments(Map options) {
+    addArguments((List<String>) options.get("args"));
+    deleteDuplicates((ArrayList) args);
+  }
+
+  private void mergeExtensions(Map options) {
+    if (options.get("extensions") != null) {
```

Comment:
Placing the entire method into a block makes it easy to miss things. Either exit quickly (by checking for `null`, and returning), or use `options.getOrDefault()` with a sensible value.